From: Swaraj Gaikwad <swarajgaikwad1925@gmail.com>
Date: Tue, 21 Oct 2025 02:17:13 +0000
Subject: [PATCH] mm/damon/sysfs: Remove misleading todo comment in nid_show()

The TODO comment in nid_show() suggested returning an error if the goal
was not using nid. However, this comment was found to be inaccurate and
misleading.  This patch removes the TODO comment without changing any
existing behavior.

This change follows feedback from SJ who pointed out [1] that writing-order
independence is expected and the function should simply show the last
set value.

No functional code changes were made.

Tested with KUnit:
- Built kernel with KUnit and DAMON sysfs tests enabled.
- Executed KUnit tests:
  ./kunit.py run --kunitconfig ./mm/mm/damon/tests/.kunitconfig
- All 25 tests passed, including damon_sysfs_test_add_targets.

[1] https://lore.kernel.org/20251020151315.66260-1-sj@kernel.org

Signed-off-by: Swaraj Gaikwad <swarajgaikwad1925@gmail.com>
Suggested-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20251021021712.59017-2-swarajgaikwad1925@gmail.com
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs-schemes.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/damon/sysfs-schemes.c b/mm/damon/sysfs-schemes.c
index c98cf4bd2fbb..30d20f5b3192 100644
--- a/mm/damon/sysfs-schemes.c
+++ b/mm/damon/sysfs-schemes.c
@@ -1121,7 +1121,6 @@ static ssize_t nid_show(struct kobject *kobj,
 	struct damos_sysfs_quota_goal *goal = container_of(kobj, struct
 			damos_sysfs_quota_goal, kobj);
 
-	/* todo: return error if the goal is not using nid */
 
 	return sysfs_emit(buf, "%d\n", goal->nid);
 }
-- 
2.47.3

